sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuOverrides.scala

```diff
@@ -2372,7 +2373,8 @@ object GpuOverrides {
       (agg, conf, p, r) => new GpuSortAggregateMeta(agg, conf, p, r)),
     exec[SortExec](
       "The backend for the sort operator",
-      ExecChecks(TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL, TypeSig.all),
+      ExecChecks((TypeSig.commonCudfTypes + TypeSig.NULL + TypeSig.DECIMAL + TypeSig.ARRAY +
+        TypeSig.STRUCT).nested(), TypeSig.all),
```

Comment:
Technically this allows arrays of arrays, arrays of structs, and decimals in there too. I don't have a problem with that so long as we have tested it.

We might want a comment here about `SortOrder` will restrict what we actually allow to be sorted by and other types are only shuffled.